media_driver/agnostic/gen11/codec/kernel_free/media_srcs.cmake

```diff
@@ -18,6 +18,80 @@
 # ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
 # OTHER DEALINGS IN THE SOFTWARE.
 
+function(gen_codec_kernel_from_cm)
+
+    set(genx "g11")
+    set(kind "")
+    set(name "codec")
+    set(platform "gen11")
+
+    set(krn ig${name}krn_${genx})
+    set(krn_dir ${CMAKE_SOURCE_DIR}/media_driver/agnostic/${platform}/${name}/kernel_free)
+    set(out_dir ${CMAKE_SOURCE_DIR}/media_driver/agnostic/${platform}/${name}/kernel_free/cache_kernel)
+    set(kernel_dir ${out_dir}/kernel)
+    set(krn_header ${CMAKE_SOURCE_DIR}/media_driver/agnostic/common/${name}/kernel/${name}krnheader.h)
+
+    message("krn: " ${krn})
+    message("krn_dir: " ${krn_dir})
+    message("out_dir: " ${out_dir})
+    message("kernel_dir: " ${kernel_dir})
+    message("krn_header: " ${krn_header})
+
+    add_custom_command(
+        OUTPUT ${out_dir} ${kernel_dir} ${patch_dir} ${kernel_hex_dir} ${patch_hex_dir}
+        COMMAND ${CMAKE_COMMAND} -E make_directory ${out_dir}
+        COMMAND ${CMAKE_COMMAND} -E make_directory ${kernel_dir}
+        COMMENT "Creating ${name} cmfc kernels output directory")
+
+    # Compiling all the sources in the kernel source directory.
+    file(GLOB_RECURSE srcs ${krn_dir}/Source/*.cpp)
+
+    set(objsname "")
+    set(cm_genx icllp)
+
+    foreach(src ${srcs})
+        get_filename_component(obj ${src} NAME_WE) # there are other outputs from cmc command, but we use only .dat and .fcpatch
+
+        add_custom_command(
+            OUTPUT ${out_dir}/${obj}_0.dat
+            DEPENDS ${src} ${out_dir}
+            WORKING_DIRECTORY ${out_dir}
+            COMMAND ${CMC}
+                -c -Qxcm -Qxcm_jit_target=${cm_genx} 
+                -I ${krn_dir}/Source/ 
+                -Qxcm_jit_option="-nocompaction -output -binary" 
+                -mCM_emit_common_isa 
+                -mCM_no_input_reorder 
+                #-mCM_unique_labels=MDF_FC 
+                -mCM_printregusage
+                -mCM_old_asm_name
+                ${src}
+            COMMENT "Compiling ${src}... to ${out_dir}/${obj}_0.dat\
+                Checking if renaming is needed..."
+        )
+
+        set(objsname ${objsname} ${out_dir}/${obj}_0.dat)
+    endforeach()
+
+    # Generating source from the .krn file
+    #get_c_macro_int(${CMAKE_CURRENT_LIST_DIR}/common/codec/kernel/codeckrnheader.h
+    #    "IDR_CODEC_TOTAL_NUM_KERNELS" IDR_CODEC_TOTAL_NUM_KERNELS)
+    add_custom_command(
+        OUTPUT ${krn_dir}/${krn}.c ${krn_dir}/${krn}.h
+        DEPENDS KernelBinToSource  ${objsname}
+        WORKING_DIRECTORY ${out_dir}
+        COMMAND ${PYTHON} ${CMAKE_SOURCE_DIR}//media_driver/agnostic/common/codec/kernel/merge.py -o ${krn}.krn ${objsname}
+        # ${index} is needed to match a description in the following file:
```

Comment:
what is this comment about? I don't understand how it correlates...

Reply:
Not needed after refactor. Remove it.